src/apis/auth/util/getSnsProfile.ts

```diff
@@ -0,0 +1,73 @@
+import fetch from 'node-fetch';
+import { GoogleUserResponse, KakaoUserResponse, NaverUserResponse, SnsProfileBase } from '../social/types/auth-social.type';
+import { HttpBadRequestException } from '@src/http-exceptions/exceptions/http-bad-request.exception';
+import { ERROR_CODE } from '@src/constants/error/error-code.constant';
+import { HttpInternalServerErrorException } from '@src/http-exceptions/exceptions/http-internal-server-error.exception';
+import { COMMON_ERROR_CODE } from '@src/constants/error/common/common-error-code.constant';
+
+/**
+ * SNS에서 사용자 프로필 정보를 가져온다.
+ * @param {string} loginType 'GOOGLE','KAKAO','NAVER'
+ * @param {string} snsToken SNS에서 발급한 token
+ * @returns {Promise<SnsProfileBase>} SNS profile 데이터
+ */
+export async function getSnsProfile(loginType: string, snsToken: string): Promise<SnsProfileBase> {
```

Comment:
```suggestion
export async function getSnsProfile(loginType: UserRole, snsToken: string): Promise<SnsProfileBase | null> {
```
지금 tsconfig 설정이 느슨해서 타입에러가 안나는데 `strictNullCheck: true`로 설정할 경우 에러가 나는 정의입니다.